_posts/2024-02-21-securing-wildfly-apps-okta-openshift.adoc

```diff
@@ -0,0 +1,260 @@
+
+We will need to first fork, and then clone the https://github.com/wildfly-security-incubator/elytron-examples[elytron-examples] repository. We will be using the `simple-webapp-okta` directory in this repo:
+```
+git clone git@github.com:<your_github_username>/elytron-examples.git
+cd simple-webapp-okta
+```
+
+== Log Into the OpenShift Cluster
+Before we can deploy our application, we need to log into an OpenShift cluster. You can log in via the https://docs.openshift.com/container-platform/4.14/cli_reference/openshift_cli/getting-started-cli.html[OpenShift CLI]:
+```
+oc login -u myUserName
+```
+Alternatively, you can log in using the API token:
+```
+oc login --token=myToken --server=myServerUrl
+```
+You can request the token via the *Copy Login Command* link in the OpenShift web console.
+
+If you don’t already have a project created, you can create one using:
+```
+oc new-project myProjectName
+```
+
+Once you have your environment set up with the required tools, we can move on to the next step to build and deploy our application on OpenShift.
+
+== Configure Okta 
+For this guide, we will be using Okta as out OpenID provider. In order to secure the application, we will need to access the Okta web console and register our application as a client. 
+
+. Log in to the https://www.okta.com/free-trial/[Okta dashboard] to get started. 
+. At the top right corner, click on *Admin* to be redirected to the admin dashboard. 
+. On the left hand panel, from the *Applications* drop down, click on *Applications* and then *Create App Integration*. Choose *OIDC - OpenID Connect* under *Sign-in method* and *Web Application* for *Application type* and click *Next*. 
+. Under *General Settings* add the following configurations: 
+* `App integration name:` *simple-webapp-okta*
+* `Grant type:` Leave it as is, with `Authorization Code` checked. 
+* `Sign-in redirect URIs`: Leave it as is for now. We will edit it later.
+* Under the `Assignments` tab, for `Controlled access`, select *Skip group assignment for now*. 
+* Click `Save`
+. Navigate to the *Directory* drop down on the left panel and click *People*. Click *Add person* and add a person with: 
+* First name: *Alice*
+* Last name: *Smith*
+* Email: *alice@example.org*
+* User Name: *alice@example.org* 
+* From the dropdown list for _Password_, choose _Set by admin_ and set a password that fits the password requirements. 
+* Uncheck _User must change password on first login_ and hit `Save`. For more information about how to add a user manually https://help.okta.com/en-us/content/topics/users-groups-profiles/usgp-add-users.htm[click here].
```

Comment:
Same here, bold instead of italic would be good for:

User must change password on first login
Applications
simple-webapp-okta
Assignments

Reply:
fixed!